osu.Game/Database/IModelManager.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+
+namespace osu.Game.Database
+{
+    public interface IModelManager<TModel>
```

Comment:
needs xmldoc. also this doesn't contain the correct constraints on `TModel`.

also this isn't really used anywhere. not sure we should be defining such interfaces too early.

Reply:
agree that it was early to include in this PR. it is used in the `DownloadTrackingComposite` update PR (#5001) to consume the items added/removed events.

also was (am still) unsure what the right constraints to have here at the interface level would be.

Reply:
Ah, I didn't realise that uses the interface. Is that for DI purposes?

Reply:
well its not used directly but by virtue of `IModelDownloader` inheriting it (which is used for the DI). the `ItemAdded`/`ItemRemoved` events are needed on the generic type.

Reply:
Right. Understood.